### PR TITLE
Add Unit Tests (for LaTeX conversion)

### PR DESCRIPTION
**Pull request by tecosaur**
As discussed in #426 

Uses the Mocha testing framework

### CODE REVIEW

File: test/LaTeXImportTests.js [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+                assert.equal('4', nerdamer.convertFromLaTeX('2 \\left[1+1\\right]').toTeX());
+            });
+            it('\\big', () => {
+                assert.equal('4', nerdamer.convertFromLaTeX('2 \\big[1+1\\big]').toTeX());
+            });
+            it('\\bigg', () => {
+                assert.equal('4', nerdamer.convertFromLaTeX('2 \\bigg[1+1\\bigg]').toTeX());
+            });
+            it('\\Big', () => {
+                assert.equal('4', nerdamer.convertFromLaTeX('2 \\Big[1+1\\Big]').toTeX());
```
Comment (Happypig375) [inline test/LaTeXImportTests.js]: Square brackets indicate vectors, so they should be treated as such.
```suggestion
                assert.equal('\\left[4\\right]', nerdamer.convertFromLaTeX('2 \\Big[1+1\\Big]').toTeX());
```

File: test/LaTeXImportTests.js [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+            });
+            it('\\Bigg', () => {
+                assert.equal('4', nerdamer.convertFromLaTeX('2 \\Bigg[1+1\\Bigg]').toTeX());
```
Comment (Happypig375) [inline test/LaTeXImportTests.js]: Square brackets indicate vectors, so they should be treated as such.
```suggestion
                assert.equal('\\left[4\\right]', nerdamer.convertFromLaTeX('2 \\Bigg[1+1\\Bigg]').toTeX());
```

File: test/LaTeXImportTests.js [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        it('phantom', () => {
+            assert.equal('2', nerdamer.convertFromLaTeX('1 \\phantom{1cm} + \\phantom{2en} 1').toTeX());
+        });
+    });
+})
+
+describe('Alternative Function', () => {
+    context('Triganometry', () => {
+        it('sin', () => {
+            assert.equal('', nerdamer.convertFromLaTeX('\\sin \\pi/2').toTeX());
```
Comment (Happypig375) [inline test/LaTeXImportTests.js]: The result should be zero
```suggestion
            assert.equal('0', nerdamer.convertFromLaTeX('\\sin \\pi/2').toTeX());
```

File: test/LaTeXImportTests.js [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+        it('square root', () => {
+            assert.equal("2", nerdamer.convertFromLaTeX('\\sqrt{4}').toTeX());
+            assert.equal("\\sqrt{5}", nerdamer.convertFromLaTeX('\\sqrt{5}').toTeX());
+        });
+        it('cube root', () => {
+            assert.equal("2", nerdamer.convertFromLaTeX('\\sqrt[3]{8}').toTeX());
+            assert.equal("\\sqrt[3]{5}", nerdamer.convertFromLaTeX('\\sqrt[3]{5}').toTeX());
+        });
+        it('nth root', () => {
+            assert.equal("\\sqrt[n]{2", nerdamer.convertFromLaTeX('\\sqrt[n]{2}').toTeX());
```
Comment (Happypig375) [inline test/LaTeXImportTests.js]: Is a close brace missing?
```suggestion
            assert.equal("\\sqrt[n]{2}", nerdamer.convertFromLaTeX('\\sqrt[n]{2}').toTeX());
```

Comment (tecosaur) [reply test/LaTeXImportTests.js]: Yes it is!

File: test/LaTeXImportTests.js (hunk 3 above)
Comment (tecosaur) [reply test/LaTeXImportTests.js]: Actually, that block is redundent and should be removed.

**Comment by Happypig375** [comment]
Currently nerdamer sees all square brackets as vectors, as other notations are unsupported.

If all notations you mentioned are to be supported, then what should nerdamer interpret (1, 1) as? A set? A vector? A range? How do you differentiate the vector `[2(2+3)]` from the integer `[2(2+3)]`?

**Comment by tecosaur** [comment]
Correct me if I'm wrong but I didn't think nerdamer supported anything with sets, so would the following approach work?

1. Get string enclosed by brackets/parenthesis
2. See if there's a comma
3. If there is, treat as vector
4. Else, treat as number

**Comment by jiggzson** [comment]
I didn't realize there was such a lively discussion going on :laughing:. I haven't yet had a chance to review the beginning of the thread but I did want to respond to 2 things. 

1. As @Happypig375 pointed out, square brackets strictly represent vectors at the moment. The only other way to create a vector is with the built-in `vector` function.
2. Round brackets with comma separated values internally map to a `Collection` object. It's currently somewhat of a loose end. Sets aren't currently supported although I hope to in the near future.

>Correct me if I'm wrong but I didn't think nerdamer supported anything with sets, so would the following approach work?

>Get string enclosed by brackets/parenthesis
See if there's a comma
If there is, treat as vector
Else, treat as number

I'm guessing that this excludes functions such as `max`. Or are we strictly speaking about stand-alone brackets?  I really think you would save yourself a lot of work by at least re-using the existing tokenizer. Brackets come back as Arrays and have a type property attached, square, curly, or round. Just a suggestion.

**Comment by jiggzson** [comment]
Can you let me know when this PR is ready? @tecosaur, if it's not too much trouble, can this and future PR's be into `dev`. I have some changes on that branch so I'd appreciate that. Or at the very least pull in those changes. I've learned through @Happypig375 that when you "Squash and Merge" the contributor data is lost so I won't be doing that anymore when merging into `master`.

**Comment by tecosaur** [comment]
@jiggzson The purpose of this PR is **(a)** to provide **a** test suite for the LaTeX import functionality, and **(b)** so that you know what I'm talking about when I say test this and that are currently passing, and that and this are failing.

I've recently discovered the jasmine tests that are here, so I'd like you to pick from one of four options
exist, so I'm going to convert the mocha tests to jasmine.

Would you like me to cancel this PR, and make another after I've rewritten the tests (to the `dev` branch this time)?

Conversation can continue in #426

**Comment by tecosaur** [comment]
Ok then. Will do :)

**Pull request closed without merging**